File: Base/GameSpec.StoreManager/StoreManager.cs
using GameSpec.StoreManagers;
using System;
using System.Text.Json;

namespace GameSpec
{
    /// <summary>
    /// StoreManager
    /// </summary>
    public static class StoreManager
    {
        public static bool TryGetPathByKey(string key, JsonProperty prop, JsonElement? keyElem, out string path)
        {
            var parts = key.Split(':', 2);
            return parts[0] switch
            {
                "Steam" => SteamStoreManager.TryGetPathByKey(parts[1], prop, keyElem, out path),
                "GOG" => GogStoreManager.TryGetPathByKey(parts[1], prop, keyElem, out path),
                "Blizzard" => BlizzardStoreManager.TryGetPathByKey(parts[1], prop, keyElem, out path),
                "Epic" => EpicStoreManager.TryGetPathByKey(parts[1], prop, keyElem, out path),
                "Unknown" => UnknownStoreManager.TryGetPathByKey(parts[1], prop, keyElem, out path),
                _ => throw new ArgumentOutOfRangeException(nameof(key), parts[0]),
            };
        }
    }
}

File: Applications/Book/book/03-locate-files/python/StoreManager.py
from BlizzardStoreManager import blizzardAppPaths
from EpicStoreManager import epicAppPaths
from GogStoreManager import gogAppPaths
from SteamStoreManager import steamAppPaths

@staticmethod
def getPathByKey(key):
    (k,v) = key.split(':', 2)
    match k:
        case 'Steam': return steamAppPaths[v] if v in steamAppPaths else None
        case 'GOG': return gogAppPaths[v] if v in gogAppPaths else None
        case 'Blizzard': return blizzardAppPaths[v] if v in blizzardAppPaths else None
        case 'Epic': return epicAppPaths[v] if v in epicAppPaths else None
        case 'Unknown': return None
        case _: raise Exception(f'Unknown key: {key}')

# print(getPathByKey('Steam:1755910'))
